### PR TITLE
Make allocator limit available for the runtime

### PR DESCRIPTION
**Pull request by thiolliere**
Fix https://github.com/paritytech/substrate/issues/9392

This move the allocator max allocation to the primitives and use it in the client while testing for the value.

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

File: primitives/core/src/sandbox.rs [hunk 1]
```diff
@@ -40,6 +40,11 @@ pub enum ExternEntity {
 	Memory(u32),
 }
 
+/// The maximum number of bytes that can be allocated at one time.
+// The maximum possible allocation size was chosen rather arbitrary, 32 MiB should be enough for
+// everybody.
+pub const MAX_POSSIBLE_ALLOCATION: u32 = 33554432; // 2^25 bytes, 32 MiB
```
Comment (bkchr) [inline primitives/core/src/sandbox.rs:46]: Why do you put this here under sandbox?

Comment (thiolliere) [reply primitives/core/src/sandbox.rs:46]: I was thinking sanbox was a good place to put the constraint of the execution of the wasm runtime.

But thinking again this seems wrong, the constraint is on the runtime.
The runtime must not do an allocation of more than 32MiB.

But I don't know where I should put this constraint, would putting it in primitives/core/src/lib.rs be fine ?

Comment (bkchr) [reply primitives/core/src/sandbox.rs:46]: Probably better than keeping it in sandbox.

**Review by kianenigma — APPROVED** [review]
thanks for `test_n_orders`, will prevent any issues.

File: primitives/core/src/sandbox.rs (hunk 1 above)
Comment (thiolliere) [reply primitives/core/src/sandbox.rs]: yes it is now moved in primitives/core/src/lib.rs

**Review by bkchr — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**